stdlib/public/runtime/ImageInspectionCOFF.cpp

```diff
@@ -35,16 +37,109 @@ void record(const swift::MetadataSections *sections) {
     registered->prev = sections;
   }
 }
+
+  // Returns true if all bytes in the given memory range are 0.
+  bool areAllBytesZero(const char *start, std::size_t length) {
```

Comment:
This is just `std::all_of` checking if the bytes are `\0`.  You can probably just delete this and use that as the branch predicate in both places this is used.